src/Relude/Extra/Validation.hs

```diff
@@ -39,6 +39,24 @@ instance Functor (Validation e) where
     _ <$ Failure e = Failure e
     {-# INLINE (<$) #-}
 
+{- | __Examples __
+>>> let a = Success "First success." :: Validation [Text] Text
+>>> let b = Success " Second success." :: Validation [Text] Text
+>>> let c = Failure ["Not correct"] :: Validation [Text] Text
+
+>>> a <> b
+Success "First success. Second success."
+
+>> a <> c
+Failure ["Not correct"]
+-}
+
+instance (Semigroup e, Semigroup a) => Semigroup (Validation e a) where
+    x <> y = (<>) <$> x <*> y
```

Comment:
I would love to have several changes in this implementation:

1. Add explicit function signature in the spirit of all instances.
2. Use `liftA2`. Basically, this function can be implemented simpler:
    ```haskell
    (<>) = liftA2 (<>)
    ```
3. Add `{-# INLINE #-}` pragmas to `<>` and `mempty`

Reply:
Sorry, I'll change this as well.